bitswap/client/internal/notifications/notifications.go

```diff
@@ -118,12 +129,14 @@ func (ps *impl) Subscribe(ctx context.Context, keys ...cid.Cid) <-chan blocks.Bl
 				if !ok {
 					return
 				}
-				block, ok := val.(traceability.Block)
-				if !ok {
-					// FIXME: silently dropping errors wtf ?
-					return
+				var block blocks.Block
+				if ps.trace {
+					tb := val.(traceability.Block)
+					tb.Delay = time.Since(subscribe)
+					block = tb
```

Comment:
Any reason not to use the following?

```suggestion
					block = val.(traceability.Block)
					block.Delay = time.Since(subscribe)
```

Reply:
It is necessary because `block` does not have a `Delay` so I have to assign the delay when the left-hand-side is a `traceability.Block` and not a `blocks.Block`